indra/llui/lllayoutstack.cpp

```diff
@@ -843,16 +847,21 @@ void LLLayoutStack::updatePanelRect( LLLayoutPanel* resized_panel, const LLRect&
                     ? new_rect.getWidth()
                     : new_rect.getHeight();
     S32 delta_panel_dim = new_dim - resized_panel->getVisibleDim();
-    if (delta_panel_dim == 0) return;
+    if (delta_panel_dim == 0)
+        return;
 
     F32 total_visible_fraction = 0.f;
     F32 delta_auto_resize_headroom = 0.f;
     F32 old_auto_resize_headroom = 0.f;
 
-    LLLayoutPanel* other_resize_panel = NULL;
-    LLLayoutPanel* following_panel = NULL;
+    LLLayoutPanel* other_resize_panel = nullptr;
+    LLLayoutPanel* following_panel = nullptr;
 
-    BOOST_REVERSE_FOREACH(LLLayoutPanel* panelp, mPanels) // Should replace this when C++20 reverse view adaptor becomes available...
+#if defined(__GNUC__) && (__GNUC__ < 13)
+    BOOST_REVERSE_FOREACH(LLLayoutPanel * panelp, mPanels)
+#else
+    for (auto panelp : mPanels | std::views::reverse)
+#endif
```

Comment:
Woof. I see the necessity, but it's too bad to have to clutter the loop source inline. On the other hand, hiding the difference in a macro would eliminate the point of using the new ranges syntax.

Trivial tweak: does `BOOST_REVERSE_FOREACH()` work with `auto` ?

Reply:
Could also try using `boost::adaptors::reversed` from `boost/range/adaptors.hpp`

` for (auto panelp : mPanels | boost::adaptors::reversed)`

Reply:
Well... yeah - but then we replace one boost construct with another which wasn't the original idea.